### PR TITLE
Refactor self join

### PR DESCRIPTION
**Pull request by SaVoAMP**
# Pull Request Checklist

Below is a simple checklist but please do not hesitate to ask for assistance!

- [x] Fork, clone, and checkout the newest version of the code
- [x] Create a new branch
- [x] Make necessary code changes
- [x] Install `black` (i.e., `python -m pip install black` or `conda install -c conda-forge black`)
- [x] Install `flake8` (i.e., `python -m pip install flake8` or `conda install -c conda-forge flake8`)
- [x] Install `pytest-cov` (i.e., `python -m pip install pytest-cov` or `conda install -c conda-forge pytest-cov`)
- [x] Run `black .` in the root stumpy directory
- [x] Run `flake8 .` in the root stumpy directory
- [x] Run `./setup.sh && ./test.sh` in the root stumpy directory
- [x] Reference a Github issue (and create one if one doesn't already exist)

### CODE REVIEW

**Comment by seanlaw** [comment]
@SaVoAMP It looks like you don't have the latest version merged with your main branch (and then into this development branch). I know this because your version `aamped` and `stumped` files have additional lines that have been removed from `main`. Can you please try merging the latest changes from main and push again?

**Comment by SaVoAMP** [comment]
I don't know what I'm doing wrong, my forked repository says that everything is up to date and there is nothing to merge. What did I miss?
image

Would reforking the repository help?

**Comment by NimaSarajpoor** [comment]
@SaVoAMP 

This is just my 2 cents :) 

Have you tried `git pull origin main` in your local repo?  So, after `fetch and merge` on github (which update YOUR "REMOTE" repo), you need to bring all those changes (from YOUR "REMOTE" repo) to your local repo as well.

```
# in git bash
git checkout main
git pull origin main # now your main branch on your local repo is up to date
git checkout YOUR_BRANCH
git merge main  
```

If you get `merge conflict` and it was hard for you to resolve it, then instead of the step `git checkout YOUR_BRANCH`, you can create a new branch and then submit a new Pull Request. 

-----

**Comment by seanlaw** [comment]
Ohh, wait! @SaVoAMP Did you simply stash the files and then copy them back over? If so, that explains it!

File: stumpy/aamped.py [hunk 1]
```diff
@@ -150,9 +150,16 @@ def aamped(dask_client, T_A, m, T_B=None, ignore_trivial=True, p=2.0):
     out[:, 0] = profile[:, 0]
     out[:, 1:4] = indices
 
-    threshold = 10e-6
-    if core.are_distances_too_small(out[:, 0], threshold=threshold):  # pragma: no cover
-        logger.warning(f"A large number of values are smaller than {threshold}.")
-        logger.warning("For a self-join, try setting `ignore_trivial = True`.")
+    # Delete data from Dask cluster
```
Comment (seanlaw) [inline stumpy/aamped.py:153]: Please delete lines 153-162

File: stumpy/stumped.py [hunk 2]
```diff
@@ -263,9 +263,24 @@ def stumped(dask_client, T_A, m, T_B=None, ignore_trivial=True, normalize=True,
     out[:, 0] = profile[:, 0]
     out[:, 1:4] = indices
 
-    threshold = 10e-6
-    if core.are_distances_too_small(out[:, 0], threshold=threshold):  # pragma: no cover
-        logger.warning(f"A large number of values are smaller than {threshold}.")
-        logger.warning("For a self-join, try setting `ignore_trivial = True`.")
+    # Delete data from Dask cluster
```
Comment (seanlaw) [inline stumpy/stumped.py:266]: Please delete lines 266-283

**Comment by SaVoAMP** [comment]
Thank you and sorry for the inconvenience!

**Comment by seanlaw** [comment]
@SaVoAMP Thank you for this PR!

**Pull request merged**